ple/games/donkeykong/Board.py

```diff
@@ -0,0 +1,335 @@
+        '''
+        self.map = []
+        # These are the arrays in which we store our instances of different classes
+        self.Players = self.Enemies = self.Allies = self.Coins = self.Walls = self.Ladders = self.Fireballs = self.Hearts = self.Boards = self.FireballEndpoints = []
+
+        # Resets the above groups and initializes the game for us
+        self.resetGroups()
+
+        self.myfont = pygame.font.SysFont("comicsansms", 50)
+
+        self.background = pygame.image.load('Assets/background.png')
+        self.background = pygame.transform.scale(self.background, (width, height))
+
+        # Initialize the instance groups which we use to display our instances on the screen
+        self.fireballGroup = pygame.sprite.RenderPlain(self.Fireballs)
+        self.playerGroup = pygame.sprite.RenderPlain(self.Players)
+        self.enemyGroup = pygame.sprite.RenderPlain(self.Enemies)
+        self.wallGroup = pygame.sprite.RenderPlain(self.Walls)
+        self.ladderGroup = pygame.sprite.RenderPlain(self.Ladders)
+        self.coinGroup = pygame.sprite.RenderPlain(self.Coins)
+        self.allyGroup = pygame.sprite.RenderPlain(self.Allies)
+        self.fireballEndpointsGroup = pygame.sprite.RenderPlain(self.FireballEndpoints)
+        self.boardGroup = pygame.sprite.RenderPlain(self.Boards)
+        self.heartGroup = pygame.sprite.RenderPlain(self.Hearts)
+
+    def resetGroups(self):
+        self.score = 0
+        self.map = []  # We will create the map again when we reset the game
+        self.Players = [Player(pygame.image.load('Assets/still.png'), (50, 440), self.rng)]
+        self.Enemies = [Donkey(pygame.image.load('Assets/kong0.png'), (100, 117), self.rng)]
+        self.Allies = [Person(pygame.image.load('Assets/princess.png'), (50, 55), self.rng)]
+        self.Allies[0].updateWH(self.Allies[0].image, "H", 0, 25, 25)
+        self.Coins = []
+        self.Walls = []
+        self.Ladders = []
+        self.Fireballs = []
+        self.Hearts = [OnBoard(pygame.image.load('Assets/heart.png'), (730, 490), self.rng),
+                       OnBoard(pygame.image.load('Assets/heart.png'), (750, 490), self.rng),
+                       OnBoard(pygame.image.load('Assets/heart.png'), (770, 490), self.rng)]
+        self.Hearts[0].modifySize(pygame.image.load('Assets/heart.png'), 20, 20)
+        self.Hearts[1].modifySize(pygame.image.load('Assets/heart.png'), 20, 20)
+        self.Hearts[2].modifySize(pygame.image.load('Assets/heart.png'), 20, 20)
+        self.Boards = [OnBoard(pygame.image.load('Assets/board.png'), (200, 480), self.rng),
+                       OnBoard(pygame.image.load('Assets/board.png'), (685, 480), self.rng)]
+        self.Boards[0].modifySize(self.Boards[0].image, 40, 150)  # Do this on purpose to get a pixelated image
+        self.Boards[1].modifySize(self.Boards[1].image, 40, 150)
+        self.FireballEndpoints = [OnBoard(pygame.image.load('Assets/still.png'), (50, 440), self.rng)]
```

Comment:
performance: might be useful to use image.convert() or image.convert_alpha() depending on if the image uses alpha